pkg/logs/parser/parser.go

```diff
@@ -0,0 +1,268 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+package parser
+
+import (
+	"bytes"
+	"errors"
+)
+
+// Length of the docker message header.
+// See https://godoc.org/github.com/moby/moby/client#Client.ContainerLogs:
+// [8]byte{STREAM_TYPE, 0, 0, 0, SIZE1, SIZE2, SIZE3, SIZE4}[]byte{OUTPUT}
+const dockerHeaderLength = 8
+
+// Docker splits logs that are larger than 16Kb
+// https://github.com/moby/moby/blob/master/daemon/logger/copier.go#L19-L22
+const dockerBufferSize = 16 * 1024
+
+const (
+	// Containerd log partial flags
+	logTagPartial = 'P'
+	logTagFull    = 'F'
+	// Containerd log stream type
+	stdout = "stdout"
+	stderr = "stderr"
+)
+
+// ParsedLine represents a containerd message
+type ParsedLine struct {
+	Content   []byte
+	Severity  string
+	Timestamp string
+}
+
+// Parser parse containerd messages
+type Parser interface {
+	Parse([]byte) (ParsedLine, error)
+	Unwrap([]byte) ([]byte, error)
+}
+
+// IdentityParser is the default parser and does nothing
+type IdentityParser struct{}
+
+// NewIdentityParser returns a new IdentityParser
+func NewIdentityParser() *IdentityParser {
+	return &IdentityParser{}
+}
+
+// Parse does nothing for IdentityParser
+func (p *IdentityParser) Parse(msg []byte) (ParsedLine, error) {
+	return ParsedLine{Content: msg, Severity: ""}, nil
+}
+
+// Unwrap does nothing for IdentityParser
+func (p *IdentityParser) Unwrap(msg []byte) ([]byte, error) {
+	return msg, nil
+}
+
```

Comment:
Would it make sense to move that to `parser/containerd.go` and `parser/docker.go` ?